File: Graph/dist_of_nearest_1_bfs.java
import java.util.*;
class Solution {
    public ArrayList<ArrayList<Integer>> nearest(int[][] grid) {
        int n = grid.length;
        int m = grid[0].length;
        ArrayList<ArrayList<Integer>> res = new ArrayList<>(n);
        for(int i=0; i<n; i++){
            res.add(new ArrayList<>(m));
            for(int j=0; j<m; j++){
                res.get(i).add(0);
            }
        }
        boolean[][] vis = new boolean[n][m];
        Queue<int[]> q = new LinkedList<>();
        for(int i=0; i<n; i++){
            for(int j=0; j<m; j++){
                if(grid[i][j]==1){
                    vis[i][j]=true;
                     res.get(i).set(j, 1);
                    q.add(new int[]{i,j,0});
                }
            }
        }
        int[] delrow = {-1,0,1,0};
        int[] delcol = {0,1,0,-1};
        while(!q.isEmpty()){
            int[] top = q.poll();
            int r=top[0]; 
            int c=top[1];
            int t=top[2];
             res.get(r).set(c, t);
            for(int i=0; i<4; i++){
                int nrow=r+delrow[i];
                int ncol=c+delcol[i];
                if(nrow>=0 && nrow<n && ncol>=0 && ncol<m &&!vis[nrow][ncol]){
                    vis[nrow][ncol]=true;
                    q.add(new int[]{nrow,ncol,t+1});
                }
            }
        }
        return res;
    }
}

File: Graph/dist_of_nearest_1_bfs.cpp
vector<vector<int>> nearest(vector<vector<int>>& grid) {
        int n = grid.size(), m = grid[0].size();
        vector<vector<int>> adj(n, vector<int>(m, -1));
        queue<pair<pair<int,int>,int>> q;
        vector<vector<bool>> vis(n, vector<bool>(m, false)); 
        
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < m; j++) {
                if(grid[i][j] == 1) {
                   adj[i][j] = 0;
                   vis[i][j] = true;
                   q.push({{i, j}, 0});
                }
            }
        }
        
        int delrow[] = {-1, 0, 1, 0};
        int delcol[] = {0, 1, 0, -1};
        
        while(!q.empty()) {
            int row = q.front().first.first;
            int col = q.front().first.second;
            int tm = q.front().second;
            q.pop();
            adj[row][col] = tm;
            
            for(int i = 0; i < 4; i++) {
                int nrow = row + delrow[i];
                int ncol = col + delcol[i];
                if(nrow >= 0 && ncol >= 0 && nrow < n && ncol < m && !vis[nrow][ncol]) { 
                    vis[nrow][ncol] = true;
                    q.push({{nrow, ncol}, tm + 1}); 
                }
            }
        }
        return adj;
    }